components/opc-ua/src/Microsoft.Azure.IIoT.OpcUa.Edge.Publisher/src/Engine/LegacyJobOrchestrator.cs

```diff
@@ -308,6 +320,7 @@ IJsonSerializer jsonSerializer
                         }
                         else {
                             availableJobs.Add(newJobId, newJob);
+                            _publisherDiagnosticInfo.Add(newJobId, new JobDiagnosticInfoModel());
```

Comment:
Why are we adding a new entry to `_publisherDiagnosticInfo` in this case ? Since this job is going to end up in `_availableJobs` list that means that there is no worker that has picked it up yet and started processing it. So I would say that there should not be an entry for it present in `_publisherDiagnosticInfo` just yet, only after it has been picked up.

Reply:
removed